presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -478,6 +492,66 @@ protected Node visitShowCreate(ShowCreate node, Void context)
             throw new UnsupportedOperationException("SHOW CREATE only supported for tables and views");
         }
 
+        @Override
+        protected Node visitShowCreateFunction(ShowCreateFunction node, Void context)
+        {
+            QualifiedFunctionName functionName = qualifyFunctionName(node.getName());
+            Collection<? extends SqlFunction> functions = metadata.getFunctionManager().getFunctions(session.getRequiredTransactionId(), functionName);
```

Comment:
Hmm, isn't it a little extreme to require transaction to run SHOW CREATE FUNCTION? It's probably ok to just fetch some version?

Reply:
If you want to require transaction for sql functions, make the error message explicit at least.

Reply:
@rongrong
For 1: Transaction ID is guaranteed to exist during query execution, even for any single statement query.
For 2: If Transaction ID is missing, we're having an internal error here. The `getRequiredTransaction` method already checks for `transactionId.isPresent()` and will throw `IllegalArgumentException`, which will be converted to a GENERIC_INTERNAL_ERROR.

Reply:
I don't think missing transaction ID should be an illegal argument exception. I also don't think it's necessary to require transaction ID to exist. The fact that it probably always exist is a different matter. I think function namespaces should still work when there's no transaction. You cannot guarantee versioning consistency, but other than that why it should fail altogether?